File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/chain_of_responsibility/lab/CustomerServiceSupervisor.java
// CustomerServiceSupervisor.java

public class CustomerServiceSupervisor
{
	public boolean canHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		return (aCustomerRequestsDiscountEvent.getPercent() < 20);
	}

	public void doHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		System.out.println("I am a CustomerServiceSupervisor.");
		System.out.println("I can give you a " + aCustomerRequestsDiscountEvent.getPercent() + "% discount.");
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/chain_of_responsibility/lab/customer_service_supervisor.cpp
// customer_service_supervisor.cpp

#include <iostream>
#include "customer_requests_discount_event.h"
#include "customer_service_supervisor.h"

customer_service_supervisor::customer_service_supervisor()
{
}

bool customer_service_supervisor::can_handle(customer_requests_discount_event * a_customer_requests_discount_event) const
{
	return (a_customer_requests_discount_event->get_percent() < 20);
}

void customer_service_supervisor::do_handle(customer_requests_discount_event * a_customer_requests_discount_event)
{
	std::cout << "I am a customer_service_supervisor." << std::endl;
	std::cout << "I can give you a " << a_customer_requests_discount_event->get_percent() << "% discount." << std::endl;
}
